### PR TITLE
Update for OkHttp3

### PR DESCRIPTION
**Pull request by rogerhu**
Fixes for OkHttp3

### CODE REVIEW

File: Parse/src/main/java/com/parse/ParseOkHttpClient.java [hunk 1]
```diff
@@ -48,17 +47,18 @@
...
 
     // Don't handle redirects. We copy the setting from AndroidHttpClient.
     // For detail, check https://quip.com/Px8jAxnaun2r
-    okHttpClient.setFollowRedirects(false);
+    builder.followRedirects(false);
 
-    okHttpClient.setSslSocketFactory(SSLCertificateSocketFactory.getDefault(
-        socketOperationTimeout, sslSessionCache));
+    // Broken until OkHttp3 can expose trust managers -- https://github.com/square/okhttp/commit/85f74e2004eaf0d4ff339e8644df3d8e716361e5
+    // builder.sslSocketFactory(SSLCertificateSocketFactory.getDefault(socketOperationTimeout, sslSessionCache));
```
Comment (rogerhu) [inline Parse/src/main/java/com/parse/ParseOkHttpClient.java]: is this still necessary given the connect timeout is already setup?

Comment (grantland) [inline Parse/src/main/java/com/parse/ParseOkHttpClient.java]: I'm not entirely sure, but I think this was a relic from our original use of `AndroidHttpClient.newInstance(userAgent, sessionCache)`.

If you are sure that OkHttp client doesn't need a SSL cache, I'd be happy to accept this!

Comment (rogerhu) [inline Parse/src/main/java/com/parse/ParseOkHttpClient.java]: It doesn't need one except helps obviously with saving SSL handshakes when reconnecting.

In order to use it, I think we have to wait for OkHttp v3.3.0 to drop.  See conversation here:

https://github.com/square/okhttp/issues/2323#issuecomment-207231855

Comment (rogerhu) [inline Parse/src/main/java/com/parse/ParseOkHttpClient.java]: I don't have enough experience to know whether SSL caching functionality is a blocker... anyone on the Parse team could provide some insight?

**Comment by setheclark** [comment]
I don't see a change in the ParseHttpClient class to update the OKHTTPCLIENT_PATH field for OkHttp3.  I think this means the classpath lookup to make sure OkHttp3 in available would fail.

**Comment by rogerhu** [comment]
Thanks for checking.  Updated.

**Comment by facebook-github-bot-6** [comment]
@rogerhu updated the pull request.

File: Parse/src/main/java/com/parse/ParseOkHttpClient.java (hunk 1 above)
Comment (grantland) [inline Parse/src/main/java/com/parse/ParseOkHttpClient.java]: I think it would be safe to use the default, so we can just remove this bit.

**Comment by grantland** [comment]
Mind removing that commented out bit? After that it LGTM!

**Comment by rogerhu** [comment]
Remove commented section

**Comment by facebook-github-bot-8** [comment]
@rogerhu updated the pull request.

**Comment by grantland** [comment]
Merging. Thanks for the contribution!

**Comment by rogerhu** [comment]
Thanks!  Any chance to cut a release?

**Comment by grantland** [comment]
Currently waiting on one PR, but will make a cut next week if it isn't in by then.

**Pull request merged**